client/src/main/java/org/eclipse/hono/client/impl/AbstractRequestResponseClient.java

```diff
@@ -323,17 +347,16 @@ protected abstract R getResult(
             ApplicationProperties applicationProperties);
 
     /**
-     * The target address of the sender link on which the request message is sent.
+     * The default target address for request messages sent with this client.
      * <p>
-     * This default implementation returns the target address of the request message.
+     * This default implementation returns the link target address.
      * <p>
-     * Subclasses may override this method in order to use a different address for creating the sender link.
-     * For example, an empty address could be returned here to create an anonymous relay link.
+     * Subclasses may override this method in order to use a different address as default for sending messages.
      *
-     * @return The link target address.
+     * @return The message target address.
      */
-    protected String getLinkTargetAddress() {
-        return targetAddress;
+    protected String getDefaultMessageTargetAddress() {
```

Comment:
FMPOV this should be named *getMessageTargetAddress*. By default, it returns the link target address, but that can be changed by subclasses, right? However, this method is used to determine the value to put into the *to* property of the request message, right?

Reply:
I've put the `default` in the name because this address is not *necessarily* used when sending a message with the client.
`AbstractRequestResponseClient` has 2 `createAndSendRequest` methods now; one with and one without a `requestTargetAddress` parameter. The one without uses the `getDefaultMessageTargetAddress()` address then.

Reply:
I see, you introduced a variant for sending a message using a (message) specific address for the command client which sends messages with an address that contains the device ID whereas the sender link's target address doesn't. So the default address is usually the sender link's target address, right?

Reply:
yes